[v1.9.x][License] Fix redundant ASF and Microsoft Apache 2.0 license files

sandeep-krishnamurthy opened the pull request:
## Description ##
This PR files Item 6 in this issue - https://github.com/apache/incubator-mxnet/issues/20475

1. Remove redundant ASF license header when there is Microsoft Apache 2.0 license and copyright header.
2. Remove stale copyright by authors.
3. Add relevant files with Apache 2.0 license + Microsoft copyright in LICENSE file
4. Add MIT license files (deformable conv operators) in LICENSE file. [Source repo - https://github.com/msracver/Deformable-ConvNets/tree/master/DCNv2_op/nn]

## Checklist ##
### Essentials ###
- [X] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [X] Changes are complete (i.e. I finished coding on this PR)

leezu commented on LICENSE:
```diff
@@ -284,6 +284,22 @@
 
     3rdparty/openmp
 
+    =======================================================================================
+    Apache-2.0 license + Microsoft Copyright
+    =======================================================================================
```
The License file should not contain information about copyright. Copyright notices go to NOTICE file but only when legally required: https://infra.apache.org/licensing-howto.html#mod-notice

leezu reviewed (commented):
Copyright notices shouldn't go to LICENSE AFAIK. At least it's not required and for consistency we may want to keep them out. WDYT?

Can you confirm that the reason for removing the ASF headers is that you think the files only contain minor modifications compared to the original Microsoft AL2 implementation? For major modifications, it's totally fine to add ASF header. https://www.apache.org/legal/src-headers.html#faq-exceptions

I also want to clarify that the ASF header is not redundant to the Microsoft AL2 header.  But it only makes sense for major modifications.

sandeep-krishnamurthy commented:
- Here is the PR that introduced these files - https://github.com/apache/incubator-mxnet/pull/6298/files
- No ICLA from contributors. Hence, copyright notices had to be left as is.
- However, since it is Apache 2.0 licensed, MXNet is compliant with these files.

As an additional step, I will reach to contributors, if they can file ICLA, if yes, we can have common ASF header, which is not necessary (since these files are Apache 2.0 license) but nice to have.
* [Yuwen Xiong](https://github.com/Orpine/)
* [Haozhi Qi](https://github.com/Oh233/)
* [Yi Li](https://github.com/liyi14/)
* [Guodong Zhang](https://github.com/gd-zhang/)
* [Xizhou Zhu](https://github.com/einsiedler0408/)

sandeep-krishnamurthy commented:
@leezu 
>> Copyright notices shouldn't go to LICENSE AFAIK. At least it's not required and for consistency we may want to keep them out. WDYT?

Agreed. They are Apache 2.0 license. Copyright need not come here in License file.

>> Can you confirm that the reason for removing the ASF headers is that you think the files only contain minor modifications compared to the original Microsoft AL2 implementation? For major modifications, it's totally fine to add ASF header. https://www.apache.org/legal/src-headers.html#3party

That is correct.

sandeep-krishnamurthy commented:
@mxnet-bot run ci [unix-cpu, windows-gpu]

sandeep-krishnamurthy commented:
@mxnet-bot run ci [unix-cpu]

sandeep-krishnamurthy commented:
@leezu CI is green. Requesting your review/merge.

sandeep-krishnamurthy commented:
@leezu CI is green. Requesting your review/merge.

leezu reviewed (commented):
Thank you @sandeep-krishnamurthy. I noticed that 5 of the files are licensed under MIT license by Microsoft, but they are not listed in the LICENSE file

sandeep-krishnamurthy commented:
> Thank you @sandeep-krishnamurthy. I noticed that 5 of the files are licensed under MIT license by Microsoft, but they are not listed in the LICENSE file

I don't see any MIT license files in this PR. Are you referring to other files outside this PR?

All the files in this PR are Caffe license (verified they are in LICENSE) and rest of Apache 2.0 license with Microsoft copyright (since they are Apache 2.0 license, did not explicitly specified in LICENSE file).

leezu commented:
You removed `  * Licensed under The MIT License` from 5 files in this PR. You can search the string in https://github.com/apache/incubator-mxnet/pull/20520/files

sandeep-krishnamurthy commented:
> You removed ` * Licensed under The MIT License` from 5 files in this PR. You can search the string in https://github.com/apache/incubator-mxnet/pull/20520/files

Yes. But those were not listed in License file earlier so no change w.r.t it in License file in this PR.

leezu commented:
That may be a mistake in LICENSE rather than mistaken license header in the file. The original code is at https://github.com/msracver/Deformable-ConvNets/blob/6aeda878a95bcb55eadffbe125804e730574de8d/DCNv2_op/modulated_deformable_convolution.cc and under MIT license.

sandeep-krishnamurthy commented:
> That may be a mistake in LICENSE rather than mistaken license header in the file. The original code is at https://github.com/msracver/Deformable-ConvNets/blob/6aeda878a95bcb55eadffbe125804e730574de8d/DCNv2_op/modulated_deformable_convolution.cc and under MIT license. "Do not modify or remove any copyright notices or licenses within third-party works." https://www.apache.org/legal/src-headers.html#3party

Ah! thanks for source repo. Fixed issues as below:
1. There were 6 files (deformable_*, modulated_deformable_* ops) under MIT per source repo. They are not substantially modified. Removed ASF 2.0 header in them. Maintained original MIT. Updated list of these files in LICENSE file under MIT license section.
